kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/GpsdPositionProvider.java

```diff
@@ -0,0 +1,300 @@
+                toMetersPerSecondMeasurement(internalState.get().getSpeed(), internalState.get().getSpeedError()),
+                toRadiansMeasurement(internalState.get().getCourse(), internalState.get().getCourseError()));
+    }
+
+    @Override
+    public NmeaPosition getNmeaPosition() {
+        return new NmeaPosition(internalState.get().getLatitude(), internalState.get().getLongitude(),
+                internalState.get().getAltitude(), internalState.get().getSpeed(), internalState.get().getCourse());
+    }
+
+    @Override
+    public String getNmeaTime() {
+        throw new UnsupportedOperationException("GpsdPositionProvider doesn't return NMEA time.");
+    }
+
+    @Override
+    public String getNmeaDate() {
+        throw new UnsupportedOperationException("GpsdPositionProvider doesn't return NMEA time.");
+    }
+
+    @Override
+    public LocalDateTime getDateTime() {
+        return LocalDateTime.ofEpochSecond(internalState.get().getTimestamp().longValue(), 0, ZoneOffset.UTC);
+    }
+
+    @Override
+    public boolean isLocked() {
+        if (!this.configuration.isEnabled()) {
+            return false;
+        }
+        if (this.configuration.isStatic()) {
+            return true;
+        }
+
+        ENMEAMode enmeaMode = internalState.get().getMode();
+
+        return (enmeaMode == ENMEAMode.TwoDimensional || enmeaMode == ENMEAMode.ThreeDimensional);
+    }
+
+    @Override
+    public String getLastSentence() {
+        throw new UnsupportedOperationException("GpsdPositionProvider doesn't return NMEA sentences.");
+    }
+
+    @Override
+    public void init(PositionServiceOptions configuration, Listener gpsDeviceListener,
+            GpsDeviceAvailabilityListener gpsDeviceAvailabilityListener) {
+
+        this.configuration = configuration;
+        this.internalState = new AtomicReference<>();
+        this.internalState.set(new GpsdInternalState());
+
+        this.gpsEndpoint = new GPSdEndpoint(configuration.getGpsdHost(), configuration.getGpsdPort());
+        this.gpsEndpoint.addListener(this);
+        try {
+            this.gpsEndpoint.watch(true, true);
+        } catch (IOException e) {
+            logger.info("Unable to start the Gpsd watch mode", e);
+        }
+    }
+
+    @Override
+    public PositionProviderType getType() {
+        return PositionProviderType.GPSD;
+    }
+
+    @Override
+    public void handleATT(ATTObject att) {
+        // Noting to do.
+    }
+
+    @Override
+    public void handleDevice(DeviceObject device) {
+        // Noting to do.
+    }
+
+    @Override
+    public void handleDevices(DevicesObject devices) {
+        // Noting to do.
+    }
+
+    @Override
+    public void handleSKY(SKYObject sky) {
+        // Noting to do.
+    }
+
+    @Override
+    public void handleSUBFRAME(SUBFRAMEObject subframe) {
+        // Noting to do.
+    }
+
+    @Override
+    public void handleTPV(TPVObject tpv) {
+        internalState.get().setLatitude(tpv.getLatitude());
```

Comment:
Probably we should create a new `GpsdInternalState` instance here, fill it from the `TPVObject` and then "publish" it atomically with `internalState.set()`.